File: app/src/main/java/com/travlog/android/apps/viewmodels/SignUpPasswordViewModel.java
package com.travlog.android.apps.viewmodels;

import android.support.annotation.NonNull;

import com.travlog.android.apps.FragmentViewModel;
import com.travlog.android.apps.libs.Environment;
import com.travlog.android.apps.ui.fragments.SignUpPasswordFragment;
import com.travlog.android.apps.viewmodels.inputs.SignUpPasswordViewModelInputs;
import com.travlog.android.apps.viewmodels.outputs.SignUpPasswordViewModelOutputs;

public class SignUpPasswordViewModel extends FragmentViewModel<SignUpPasswordFragment>
        implements SignUpPasswordViewModelInputs, SignUpPasswordViewModelOutputs {

    public SignUpPasswordViewModel(final @NonNull Environment environment) {
        super(environment);
    }

    public final SignUpPasswordViewModelInputs inputs = this;
    public final SignUpPasswordViewModelOutputs outputs = this;
}

File: app/src/main/java/com/travlog/android/apps/viewmodels/SignUpPasswordViewModel.kt
package com.travlog.android.apps.viewmodels

import com.travlog.android.apps.FragmentViewModel
import com.travlog.android.apps.libs.Environment
import com.travlog.android.apps.ui.fragments.SignUpPasswordFragment
import com.travlog.android.apps.viewmodels.inputs.SignUpPasswordViewModelInputs
import com.travlog.android.apps.viewmodels.outputs.SignUpPasswordViewModelOutputs
import javax.inject.Inject

class SignUpPasswordViewModel @Inject constructor(environment: Environment
) : FragmentViewModel<SignUpPasswordFragment>(environment), SignUpPasswordViewModelInputs, SignUpPasswordViewModelOutputs {

    val inputs: SignUpPasswordViewModelInputs = this
    val outputs: SignUpPasswordViewModelOutputs = this
}
